dbutils.c

```diff
@@ -1852,6 +1852,51 @@ get_wal_receiver_pid(PGconn *conn)
 /* =============================== */
 
 
+/*
+ * Determine if the user associated with the current connection can execute CHECKPOINT command.
+ * User must be a supersuer or a member of the pg_checkpoint default role (available from PostgreSQL 15).
+ */
+bool
+can_execute_checkpoint(PGconn *conn)
+{
+	PQExpBufferData query;
+	PGresult   *res;
+	bool		has_pg_checkpoint_role = false;
+
+	/* superusers can do anything, no role check needed */
+	if (is_superuser_connection(conn, NULL) == true)
+		return true;
+
+	/* pg_checkpoint available from PostgreSQL 15 */
+	if (PQserverVersion(conn) < 150000)
+		return false;
+
+	initPQExpBuffer(&query);
+	appendPQExpBufferStr(&query,
+						 " SELECT pg_catalog.pg_has_role('pg_checkpoint','MEMBER') ");
```

Comment:
Effectively is better to check for `USAGE` privilege as `MEMBER` privilege works only if `INHERIT` is enabled in the role.

However I checked in Repmgr and Barman (in which `pg_has_role` is used) source code and in all cases the check is on `MEMBER` privilege.

I think the source code should be uniform so if `USAGE` is chosen here all occurrences of `pg_has_role` in Repmgr and Barman sources codes (more or less six) must be updated.

Reply:
I will have to check in `barman` but in `repmgr` if there is no `INHERIT` of privileges, `repmgr` will not be able to use the privileges given that there is no `SET ROLE`, it just runs the commands using the inherited privileges.

Of course, this still works as long as the membership is inherited even using `USAGE` instead of `MEMBER`.

Let me check with the Barman team and some other engineers before proceeding with the change.

Reply:
I just had a discussion with @barthisrael and we should have the check done on `USAGE` instead of `MEMBER` as it seems we both agreed. If you could provide that change to the PR here we'd be able to merge it.

We'll address the issue in Barman as well as it's the same issue. We should be using `USAGE` given we don't use `SET ROLE` when running privilege queries that require certain role escalation.

Reply:
Thanks for the checks.
I changed `MEMBER` with `USAGE` here and I opened the following two PRs to fix it everywhere:

Repmgr: https://github.com/EnterpriseDB/repmgr/pull/862
Barman: https://github.com/EnterpriseDB/barman/pull/1012